parseclj-lex.el

```diff
@@ -57,6 +57,19 @@ Tokens at a mimimum have these attributes
 Other ATTRIBUTES can be given as a flat list of key-value pairs."
   (apply 'a-list :token-type type :form form :pos pos attributes))
 
+(defun parseclj-lex-error-token (pos &optional error-type)
+  "Create a lexer error token starting at POS.
+ERROR-TYPE is an optional keyword to attach to the created token,
+as the means for providing more information on the error."
+  (when (= pos (point))
+    (right-char))
```

Comment:
Can you explain what these two lines do in the docstring? Are the calls to `(right-char)` below still necessary?

Reply:
Hmm I meant it as a fail-safe, so that never happens that you return a `:lex-error`, point doesn't move, and then you get stuck forever getting that same error.  Right now the only place that this might happen is at lineno 423, so I guess it wouldn't be a problem just moving the point there and make it look like all the other errors where we're consuming one more character before returning the `:lex-error`.

Another approach would be to always `(right-char)` here, but I don't like that side effect for this function.

> Are the calls to (right-char) below still necessary?

Yes, cause they consume the character that made the lexer stop a particular recognition. i.e. for number `123x`, the pointer is at `123|x` when the error is recognized, and you want to include the `x` in the `:lex-error` token form.

Reply:
Moving it outside of the function sounds like a good approach.